### PR TITLE
SMT should not crash on special function calls (typecast)

### LINKED ISSUE #3001: SMT Checker Error
I tried adding `pragma experimental SMTChecker;` to the top of this file:
https://github.com/AugurProject/augur-core/blob/develop/source/contracts/Controlled.sol

However, it errors with the error below when I try to compile it.

```
joeykrug:~/workspace/source/contracts (develop) $ solc Controlled.sol --bin
Unknown exception during compilation.
```

Removing the SMTChecker line it compiles fine, seems to be an issue with the SMTChecker.

### PR DESCRIPTION
**Pull request by axic**
Fixes #3001.

### CODE REVIEW

**Review by chriseth — APPROVED** [review]

**Comment by axic** [comment]
Test to be added:

```
pragma experimental SMTChecker;

contract C {
  function f() {
    uint8(1);
  }
}
```

```
pragma experimental SMTChecker;

contract C {
  struct A { uint a; uint b; }
  function f() {
    A({a: 1, b: 2 });
  }
}
```

**Comment by axic** [comment]
@chriseth added a new version of the code, I think that's the proper way?

File: libsolidity/formal/SMTChecker.cpp [hunk 1]
```diff
@@ -234,6 +234,16 @@ void SMTChecker::endVisit(BinaryOperation const& _op)
 
 void SMTChecker::endVisit(FunctionCall const& _funCall)
 {
+	solAssert(_funCall.annotation().kind != FunctionCallKind::Unset, "");
+	if (_funCall.annotation().kind != FunctionCallKind::FunctionCall)
+	{
+		m_errorReporter.warning(
+			_funCall.location(),
+			"Assertion checker does not yet implement this operator."
```
Comment (axic) [inline libsolidity/formal/SMTChecker.cpp]: Expression?

**Comment by chriseth** [comment]
This will be rewritten soon anyway.

**Comment by axic** [comment]
@chriseth ready to merge

**Comment by chriseth** [comment]
Tests fail.

**Comment by chriseth** [comment]
Tests still fail.

**Comment by axic** [comment]
@chriseth passes tests now

**Pull request merged**